kitsune/sumo/utils.py

```diff
@@ -315,21 +315,30 @@ def get_browser(user_agent):
     return browser
 
 
-def is_toll_free_number(data):
-    """Figure out if we have a toll free number."""
-    digits = filter(type(data).isdigit, data)
-    if not digits:
-        return False
-
-    if settings.TOLL_FREE_REGEX.match(digits):
-        return True
-    return False
-
-
-@timeout(seconds=settings.REGEX_TIMEOUT)
+# @timeout(seconds=settings.REGEX_TIMEOUT)
 def match_regex_with_timeout(compiled_regex, data):
     """Matches the specified regex.
 
     Adds a timeout to avoid catastrophic backtracking.
     """
     return any(compiled_regex.findall(data))
+
+
+def check_for_spam_content(data):
+    """Check for spam content in a given text.
+
+    Currently checks for:
+    - Toll free numbers
+    - Vanity toll free numbers
+    - Links in the text.
+    """
+
+    # keep only the digits in text
+    digits = filter(type(data).isdigit, data)
+    is_toll_free = settings.TOLL_FREE_REGEX.match(digits)
+
+    is_vanity_number = match_regex_with_timeout(settings.VANITY_PHONE_NUMBER_REGEX, data)
+
+    has_links = match_regex_with_timeout(settings.SIMPLE_DOMAIN_IP_REGEX, data)
+
+    return is_toll_free or is_vanity_number or has_links
```

Comment:
It would be great to return a truthy value with more information here, perhaps a dict like `{ 'is_toll_free': is_toll_free, 'is_vanity_number': ...` so that we can then surface that information in the review queue (like "This was flagged for having a toll free number") and do different things to users posting links and those posting numbers.